### DISCUSSION TITLE
Support for runtime api calls (via state_call)

### DISCUSSION THREAD

**Discussion by statictype** [paritytech/capi#218] [General]
Just curious if there is support for this or what's the plan. I'm guessing pallet RPCs will become obsolete soon-ish

**Comment by harrysolovay** [comment]
Very good question. It's already listed in the known RPC call method lookup (see [here](https://github.com/paritytech/capi/blob/b56fbc58e279f3931813a8aa91cc0b4d39a17202/known/rpc.ts#L132)). I'm extremely curious how you think we should treat this method moving forward.

Some follow-up questions:

- how would you like to see `state_call` used in Capi (aside from the raw RPC method)?
- how does the narrowest-possible typed experience look?
- what does this replace?
- limitations?

**Comment by statictype** [comment]
I was talking about the general direction in Substrate to replace RPC calls with runtime calls. For the end user of a JS library it doesn't make much difference what is used under the hood.
Some RPCs will be removed from pallets, as they are only channeling data to `state_call`. I was just curious if Capi is aware of this general direction and implements it under the hood already. Nice thing about this is that types should [come from the metadata](https://github.com/paritytech/substrate/issues/11648).
Please also see this [forum discussion](https://forum.parity.io/t/remove-rpcs-that-can-be-replaced-by-state-call/1070)

**Reply by harrysolovay** [reply]
Capi does not currently make use of `state_call` under-the-hood, although we likely should –– just opened #228.
